doc/userguide.md

```diff
@@ -138,22 +138,23 @@ type storageSrv struct {
   state *State
 }
 
-func (srv *storageSrv) Read(req *ReadRequest) *State {
+func (srv *storageSrv) Read(req *ReadRequest, out chan<- *State) {
```

Comment:
@leandernikolaus What do you think about this API change?? It will impact compatibility with previous Raft and other Gorums implementations.

Reply:
I don't mind breaking changes.

However, I'm not totally convinced that the benefits in this change outweigh the added complexity, especially for a first time user.

Reply:
These are the pros/cons that I can see:

Pros:
  * User gets control over handler's concurrency
  * Handlers can be started in the correct order, but do expensive work concurrently
  * The same API can be used for streaming calls (correctable?)
  * Fewer options in the Gorums plugin / simplifies gorums code

Cons:
  * Requires changes to existing projects
  * API is no longer similar to gRPC
  * Slight performance penalty due to starting more goroutines.